std/range/package.d

```diff
@@ -5277,24 +5278,23 @@ if ((isIntegral!(CommonType!(B, E)) || isPointer!(CommonType!(B, E)))
 
         this(Value current, Value pastLast, StepType step)
         {
-            if ((current < pastLast && step >= 0) ||
-                    (current > pastLast && step <= 0))
+            if ((current < pastLast && step > 0) ||
+                    (current > pastLast && step < 0))
             {
                 this.step = step;
                 this.current = current;
                 if (step > 0)
                 {
                     assert(unsigned((pastLast - current) / step) <= size_t.max);
-
-                    this.pastLast = pastLast - 1;
+                    // Cast below can't fail because current < pastLast
+                    this.pastLast = cast(Value) (pastLast - 1);
```

Comment:
This logic is correct, but hard to follow. 

I'm wondering if it wouldn't be better to refactor this into three if branches:

```D
if(current < pastLast && step > 0)
{
    this.step = step;
    this.current = current;
    ... // code currently inside the step > 0 branch
    this.pastLast += step; // this could probably be folded into last line
}
else if(current > pastLast && step < 0)
{
    this.step = step;
    this.current = current;
    ... // code currently inside the else branch of the inner if block
    this.pastLast += step;
}
else
{
    ... // step is 0 or current is on wrong side of pastLast
}
```